### PR TITLE
cleanup: simplify returning boolean expression in controller

### PR DESCRIPTION
**Pull request by wangyx1992**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
S1008 – Simplify returning boolean expression
https://staticcheck.io/docs/checks#S1008

#### Which issue(s) this PR fixes:
Fixes #

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/endpointslicemirroring/endpointslicemirroring_controller.go [hunk 1]
```diff
@@ -347,11 +347,7 @@ func (c *Controller) queueEndpoints(obj interface{}) {
 // This does not ensure that a corresponding Service exists with a nil selector.
 // That check should be performed separately.
 func (c *Controller) shouldMirror(endpoints *v1.Endpoints) bool {
-	if endpoints == nil || skipMirror(endpoints.Labels) || hasLeaderElection(endpoints.Annotations) {
-		return false
-	}
-
-	return true
+	return !(endpoints == nil || skipMirror(endpoints.Labels) || hasLeaderElection(endpoints.Annotations))
```
Comment (gautierdelorme) [inline pkg/controller/endpointslicemirroring/endpointslicemirroring_controller.go:350]: nit: might be more readable?
```suggestion
	return endpoints != nil && !skipMirror(endpoints.Labels) && !hasLeaderElection(endpoints.Annotations))
```

**Review by gautierdelorme — COMMENTED** [review]
/retest
flake https://github.com/kubernetes/kubernetes/issues/101275

/lgtm

File: pkg/controller/endpointslicemirroring/endpointslicemirroring_controller.go (hunk 1 above)
Comment (wangyx1992) [reply pkg/controller/endpointslicemirroring/endpointslicemirroring_controller.go]: @gautierdelorme Yeah, you're right. Thanks for your suggestion, this has been fixed already, waiting for review with pleasure.

**Comment by aojea** [comment]
I personally find it easier to read before this commit

File: pkg/controller/endpointslicemirroring/endpointslicemirroring_controller.go [hunk 2]
```diff
@@ -347,11 +347,7 @@ func (c *Controller) queueEndpoints(obj interface{}) {
 // This does not ensure that a corresponding Service exists with a nil selector.
 // That check should be performed separately.
 func (c *Controller) shouldMirror(endpoints *v1.Endpoints) bool {
-	if endpoints == nil || skipMirror(endpoints.Labels) || hasLeaderElection(endpoints.Annotations) {
-		return false
-	}
-
-	return true
+	return endpoints != nil && !skipMirror(endpoints.Labels) && !hasLeaderElection(endpoints.Annotations)
```
Comment (caesarxuchao) [inline pkg/controller/endpointslicemirroring/endpointslicemirroring_controller.go:350]: I personally feel the original code is easier to process, which has 3 less negations.

**Review by caesarxuchao — COMMENTED** [review]
Does this fix a static check that Kubernetes code base checks for?

File: pkg/controller/endpointslicemirroring/endpointslicemirroring_controller.go (hunk 2 above)
Comment (wangyx1992) [reply pkg/controller/endpointslicemirroring/endpointslicemirroring_controller.go:350]: Yeah, this fixes static check according to [https://staticcheck.io/docs/checks#S1008](url). Actually, the original code is more readable and easier to process. I will revert the modification in the second file or both of them, what do you think about? @caesarxuchao

**Comment by wangyx1992** [comment]
> I personally find it easier to read before this commit

Hi, @aojea. This PR fixes static check according to [https://staticcheck.io/docs/checks#S1008](url). But as you have metioned above, the original code is more readable and easier to process. I will revert the modification in the second file or both of them, what do you think about?

**Pull request closed without merging**